rest_api/controller/search.py

```diff
@@ -53,19 +53,33 @@ def _process_request(pipeline, request) -> QueryResponse:
     start_time = time.time()
     
     params = request.params or {}
-    params["Retriever"] = params.get("Retriever", {})
-    filters = {}
-    if "filters" in params["Retriever"]:  # put filter values into a list and remove filters with null value
-        for key, values in params["Retriever"]["filters"].items():
-            if values is None:
-                continue
-            if not isinstance(values, list):
-                values = [values]
-            filters[key] = values
-    params["Retriever"]["filters"] = filters
-    result = pipeline.run(query=request.query, params=params)
-    
+
+    # format global, top-level filters (e.g. "params": {"filters": {"name": ["some"]}})
+    if "filters" in params.keys(): _format_filters(params["filters"])
+
+    # format targeted node filters (e.g. "params": {"Retriever": {"filters": {"value"}}})
+    for key, value in params.items():
+        if "filters" in params[key].keys(): _format_filters(params[key]["filters"])
+
+    result = pipeline.run(query=request.query, params=params,debug=request.debug)
     end_time = time.time()
     logger.info({"request": request.dict(), "response": result, "time": f"{(end_time - start_time):.2f}"})
 
     return result
+
+
+def _format_filters(filters):
+    """
+    Adjust filters to compliant format in-place:
+    Put filter values into a list and remove filters with null value.
+    """
+
+    for key, values in filters.copy().items():
+        if values is None:
+            logger.warning(f"Request with deprecated filter format ('{key}: null'). "
+                           f"Remove null values from filters to be compliant with future versions")
```

Comment:
This would be nice to return to the user too (so with the response, rather than just a log) too but indeed I don't know how could it be done without manually adding a `warnings` field to the response itself. Just a though in case you know a simple way to do it.

Reply:
Agree, but I am also not aware of an easy way to do this and would not like to dive into a deeper refactoring here. We will probably keep these deprecated messages for 1-2 versions and then drop it anyway.